direct/transport/tpm.go

```diff
@@ -10,14 +10,32 @@ import (
 // TPM represents a logical connection to a TPM.
 type TPM interface {
 	Send(input []byte) ([]byte, error)
+}
+
+// TPMCloser represents a logical connection to a TPM and you can close it!!!.
+type TPMCloser interface {
+	TPM
 	Close() error
 }
 
+type tempTPM struct {
```

Comment:
Needs a better name...

Reply:
`compatTPM`? legacyTPM?

Reply:
seeing as we already have `LocalTPM`, we could also rename that to `LocalTPMCloser` and name this one `LocalTPM`

Reply:
I'll contribute to the bikeshed :)

If I understand @josephlr's out-of-band comments, we should return non-public types that happen to satisfy public interfaces. Therefore we need:

Our interfaces:
`TPM`
`TPMCloser`

Our private type names for each of these:
`localTPM` (implements both TPM and TPMCloser)
`wrappedTPM` (implements only TPM)
No wrappedTPMCloser because we shouldn't close a TPM whose readwriter we're borrowing

Don't need public type names like `WrappedTPM` or `LocalTPM` (so we can delete `LocalTPM` entirely). Just the public functions that return `TPM` or `TPMCloser` (`OpenSimulator`, `OpenTPM`, `WrapReadWriter`)

Reply:
I think Joe's comment makes the most sense to me too. :) Thank you for the suggestions